src/core/Akka.Persistence/AtLeastOnceDeliverySemantic.cs

```diff
@@ -326,18 +328,37 @@ public sealed class RedeliveryTick : INotInfluenceReceiveTimeout, IDeadLetterSup
         public AtLeastOnceDeliverySemantic(IActorContext context, PersistenceSettings.AtLeastOnceDeliverySettings settings)
         {
             _context = context;
-            _settings = settings;
             _deliverySequenceNr = 0;
+
+            RedeliverInterval = settings.RedeliverInterval;
+            RedeliveryBurstLimit = settings.RedeliveryBurstLimit;
+            WarnAfterNumberOfUnconfirmedAttempts = settings.WarnAfterNumberOfUnconfirmedAttempts;
+            MaxUnconfirmedMessages = settings.MaxUnconfirmedMessages;
         }
 
         /// <summary>
         /// Interval between redelivery attempts.
         /// 
         /// The default value can be configure with the 'akka.persistence.at-least-once-delivery.redeliver-interval'
-        /// configuration key. This method can be overridden by implementation classes to return
-        /// non-default values.
+        /// configuration key.
         /// </summary>
-        public virtual TimeSpan RedeliverInterval => _settings.RedeliverInterval;
+        public TimeSpan RedeliverInterval
+        {
+            get => _redeliverInterval;
+            set
+            {
+                if (_redeliverInterval != value)
```

Comment:
Doesn't this looks too surprising? I mean, using setter is usually considered to be simple operation with no side effect (except changing value type). The fact, that it resets schedule may be the source of problems, for example in debugging, in the future.

Reply:
It's here so that the semantic behaves "correctly", e.g. if you set a different interval it should send things in a different interval. I agree that having too much logic in setters is not great though (thought I thought it's acceptable and tried to limit the resets by checking for a change).

What do you propose? Have a method instead for changing this? Or have this read only and only accept the value via `ctor`? Or making this smarter by calculating the correct delay between the elapsed time and new interval?

Reply:
AFAIK those configuration values should be defined exactly once - when an actor is created. I'd probably need verify it somehow, but we're having a state machines here, so changing this value in different actor "state" may result in different side effects - and I'm not sure if all of them are acceptable.

Simplest way I can think of ATM to make it break the actor would be to call `RedeliverInterval = x` in actor's PostStop method after calling `base.PostStop()`, which is cancelling all scheduled messages. Any call like this one made afterwards would trigger it to schedule the message back again, but never actually cancel.

Reply:
That would not happen the way it's currently written as it checks for `_redeliverScheduleCancelable ` being active before restarting it.

```
if (_redeliverScheduleCancelable != null)
{
	_redeliverScheduleCancelable.Cancel();
	_redeliverScheduleCancelable = null;
	StartRedeliverTask();
}
```

But if you prefer not to have these mutable I can redo this PR in a different way - the values would be set via `ctor` only (at the cost of having to read the redelivery settings in two places (both AtLeastOnce actors that use this semantic).

The problem with that approach is that the `AtLeastOnceDeliverySemantic` is created in the actor's `ctor` and you should not access virtual fields inside a `ctor`, so the only way that comes in mind is specifying two `ctors` for each actor so that you can specify these parameters calling the right base `ctor`.

Reply:
Actually there is already a `ctor` that takes the redelivery settings. So perhaps we make the method non-virtual in order not to confuse the developers (as overwriting has no effect currently) and we fix the docs to say to use the overloaded `ctor` instead of overwriting?

Reply:
It would be nice, but the problem is backwards compatibility. I think, we can stay with your current approach. Any thoughts @Aaronontheweb ?

Reply:
@ondrejpialek @Horusiath 

> Actually there is already a ctor on the actors that takes the redelivery settings. So perhaps we make the properties non-virtual in order not to confuse the developers (as overwriting has no effect currently) and we fix the docs to say to use the overloaded ctor instead of overwriting?

Since this is going into the v1.4 branch and we can afford to take some more liberties with the API there - if overriding the properties already has no effect, I don't see an issue with removing the `virtual` modifier on those properties and just having developers use the constructor overload instead. I think making these settings immutable and settable through the constructor is probably the best way to go, in order to avoid having those side effects during the processing of the state machine as @Horusiath mentioned.

Can you go ahead with those changes @ondrejpialek ?